frontend/src/components/App/TopBar.tsx

```diff
@@ -48,12 +49,17 @@ export default function TopBar({}: TopBarProps) {
   const history = useHistory();
 
   function hasToken() {
-    return !!cluster ? !!getToken(cluster) : false;
+    const getTokenMethodToUse = store.getState().ui.functionsToOverride.getToken;
```

Comment:
In our conversation, what I had mentioned was that you could use the `store.getState().ui.functionsToOverride.getToken` inside the getToken(...)'s implementation. I see that maybe that will get us into trouble if plugin developers want to call the original method within their overridden method (which could end up calling itself), but maybe that's fine if we document this fact.

The reason why I think we have to really call the overridden method from the `getToken` is that developers maybe expect that this method will have the overridden behavior if any plugin overrides it.
And it also makes the whole code simpler.

Reply:
updated